[![install with bioconda](https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat-square)](http://bioconda.github.io/recipes/merfishtools/README.html)
[![Travis](https://img.shields.io/travis/merfishtools/merfishtools.svg?style=flat-square)](https://travis-ci.org/merfishtools/merfishtools)

MERFISHtools implement a Bayesian framework for accurately predicting gene or transcript expression from MERFISH data.
On top, differential expression analysis can be performed for two or multiple conditions, including credible intervals for fold change and coefficient of variation, and controlling the expected false discovery rate.

## Citation

If you use MERFISHtools, please cite our preprint

> Köster, Johannes; Brown, Myles; Liu, X. Shirley (2016): A Bayesian model for single cell transcript expression analysis on MERFISH data. figshare. https://dx.doi.org/10.6084/m9.figshare.3438662.v1

## Installation

MERFISHtools can be installed and updated easily via [Bioconda](http://bioconda.github.io/recipes/merfishtools/README.html).

## Usage

For usage instructions, please issue

    merfishtools --help

## Author
[Johannes Köster](https://johanneskoester.bitbucket.org)

## License

[MIT](https://github.com/merfishtools/merfishtools/blob/master/LICENSE.md)